### PR TITLE
Task Details, Task Edit and clean-up for main merge

### PR DESCRIPTION
**Pull request by Monti934**
# Pull Requests Checklist
 
Assignee: Monti934
 
- \[x\] code commented?
- \[x\] documentation added / updated?
- \[x\] pre-commit checks passed?
- \[x\] pipelines passed?
 
Reviewer:
 
- \[ \] are there any code smells?
- \[ \] did pipeline pass?
- \[ \] is the code well documented?

### CODE REVIEW

File: src/components/popups/InspectTask.tsx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+      const response = await api.put(
+        `/api/v1/teams/${teamId}/tasks/${task.taskId}`,
+        requestBody,
+        {
+          headers: {
+            Authorization: `${token}`,
+          },
+        }
+      );
+      //maybe remove when external api is ready
```
Comment (B1s9l) [inline src/components/popups/InspectTask.tsx:47]: This comment isn't clear to me as a reviewer? What all should be removed? Why only maybe?

File: src/components/popups/InspectTask.tsx [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+      location.reload();
+    } catch (error) {
+      //new error handling
+      setError("Failed to edit the Task");
+      if (error.response.status === 401) {
+        setError("You are not authorized to edit this Task");
+      } else if (error.response.status === 404) {
+        setError("The task you tried to edit does not exist");
```
Comment (B1s9l) [inline src/components/popups/InspectTask.tsx:55]: Does this message make sense to a user?

File: src/components/popups/InspectTask.tsx [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+        }
+      );
+      //maybe remove when external api is ready
+      location.reload();
+    } catch (error) {
+      setError("Failed to delete the Task");
+      if (error.response.status === 401) {
+        setError("You are not authorized to delete this Task");
+      } else if (error.response.status === 404) {
+        setError("The task you tried to delete does not exist");
```
Comment (B1s9l) [inline src/components/popups/InspectTask.tsx:79]: Does this message make sense to a user?

File: src/components/ui/TaskCard.tsx [hunk 4]
```diff
@@ -26,12 +39,20 @@ function TaskCard(props) {
...
+        requestBody,
+        {
+          headers: {
+            Authorization: `${token}`,
+          },
+        }
       );
     } catch (error) {
-      console.error(`Failed to Update Task error ${handleError(error)}`);
+      console.error("Error moving Task:", handleError(error));
```
Comment (B1s9l) [inline src/components/ui/TaskCard.tsx:51]: Soft error handling: do we return to the user that something is happening or rather not happening?

File: src/components/ui/TaskCard.tsx [hunk 5]
```diff
@@ -26,12 +39,20 @@ function TaskCard(props) {
...
     }
+
+    //maybe remove when external api is ready
+    location.reload();
```
Comment (B1s9l) [inline src/components/ui/TaskCard.tsx:55]: Could indeed work like this for now, external API: we're checking

**Review by B1s9l — COMMENTED** [review]
Looks good to me, some things could require some re-thinking but will be reviewed with upcoming changes such as the external API anyways

File: src/components/popups/InspectTask.tsx (hunk 1 above)
Comment (Monti934) [reply src/components/popups/InspectTask.tsx:47]: Refers to "location.reload();" right below the comment. Maybe remove when update via api

**Review by B1s9l — APPROVED** [review]

**Pull request merged**